genie-proto/src/main/proto/genie.proto

```diff
@@ -48,3 +48,86 @@ message AgentRegistrationResponse {
     bool agentAccepted = 3;
     string serverMessage = 4;
 }
+
+//
+// Job Specification APIs
+//
+service JobSpecificationService {
+    rpc resolveJobSpecification (ResolveJobSpecificationRequest) returns (JobSpecificationResponse);
+    rpc getJobSpecification (GetJobSpecificationRequest) returns (JobSpecificationResponse);
+}
+
+message Criterion {
```

Comment:
`OneOf`?

Reply:
The proposed algorithm for handling criteria in v4 doesn't enforce that only one of these is present exclusively.

For example one of the primary requested use cases by @stephen-mw is to be able to override the `UP` status of a cluster while still using the same tags. For this case the `Criterion` needs to be able to provide both `tags` and `status`

Reply:
👍